src/test/java/com/android/volley/CacheDispatcherTest.java

```diff
@@ -231,4 +234,22 @@ public void duplicateSoftExpiredCacheHit_successfulRequest() throws Exception {
         verify(mDelivery)
                 .postResponse(any(Request.class), any(Response.class), any(Runnable.class));
     }
+
+    @Test
+    public void processRequestNotifiesListener() throws Exception {
+        RequestQueue.RequestEventListener listener = mock(RequestQueue.RequestEventListener.class);
+        RequestQueue queue = new RequestQueue(mCache, mNetwork, 0, mDelivery);
+        queue.addRequestEventListener(listener);
+        mRequest.setRequestQueue(queue);
+
+        Cache.Entry entry = CacheTestUtils.makeRandomCacheEntry(null, false, false);
+        when(mCache.get(anyString())).thenReturn(entry);
+        mDispatcher.processRequest(mRequest);
+
+        verify(listener)
```

Comment:
I don't think these verify calls are validating the order. See:

https://static.javadoc.io/org.mockito/mockito-core/2.23.0/org/mockito/Mockito.html#in_order_verification